pkg/controller/signup.go

```diff
@@ -46,6 +47,50 @@ func (s *Signup) PostHandler(ctx *gin.Context) {
 	ctx.Writer.WriteHeaderNow()
 }
 
+// PostVerificationHandler creates a verification and updates a usersignup resource
+func (s *Signup) PostVerificationHandler(ctx *gin.Context) {
+	userID := ctx.GetString(context.SubKey)
+
+	// Read the Body content
+	var bodyBytes []byte
+	if ctx.Request.Body != nil {
+		bodyBytes, _ = ioutil.ReadAll(ctx.Request.Body)
+	}
+
+	m := make(map[string]string)
+	err := json.Unmarshal(bodyBytes, &m)
+	if err != nil {
+		log.Errorf(ctx, nil, "Request body could not be read")
+		ctx.AbortWithError(http.StatusInternalServerError, err)
+		return
+	}
+
+	// generate verification code
+	code, err := s.verificationService.GenerateVerificationCode()
+	if err != nil {
+		log.Errorf(ctx, nil, "verification code could not be generated")
+		ctx.AbortWithError(http.StatusInternalServerError, err)
+		return
+	}
+
+	userSignup, httpCode, err := s.signupService.PostVerification(s.config.GetVerificationDailyLimit(), m, userID, code)
+	if err != nil {
+		log.Errorf(ctx, nil, "phone verification has failed: %s", err.Error())
+		ctx.AbortWithError(httpCode, err)
+		return
+	}
+
+	err = s.verificationService.SendVerification(ctx, userSignup)
+	if err != nil {
+		log.Errorf(ctx, nil, "Verification for %s could not be sent", userID)
+		ctx.AbortWithError(http.StatusInternalServerError, err)
+		return
+	}
+
+	log.Infof(ctx, "phone verification has passed for userID %s", userID)
```

Comment:
This method does not verify the code, does it? It just sends it to the user.

Reply:
fixed here: https://github.com/codeready-toolchain/registration-service/blob/60ca7377aabe427ed8f8e3d642380b1b8396454e/pkg/controller/signup.go#L113